src/panels/lovelace/components/hui-energy-period-selector.ts

```diff
@@ -82,6 +177,15 @@ export class HuiEnergyPeriodSelector extends SubscribeMixin(LitElement) {
   private _updateDates(energyData: EnergyData): void {
     this._startDate = energyData.start;
     this._endDate = energyData.end || endOfToday();
+    this._period = isSameDay(this._startDate, this._endDate)
```

Comment:
I think that we should store "period" in the energy data so it's easier to show such buttons etc.

Reply:
I wonder even if we should store just a "start date" that combined with period = start + end.

That way we can still do -1 week or -1 month but, scaling up/down will keep us in the same 'range'

Reply:
I wanted to keep energy and all other elements as flexible as possible really, so it could handle any date range. Just the period selector is the part that is limiting now... I agree that this part is ugly, so maybe we could save the period in energy data just for this... 🤷‍♂️

Reply:
Made this logic a bit simpler now...